test/hotspot/gtest/memory/test_virtualspace.cpp

```diff
@@ -655,24 +655,28 @@ class TestVirtualSpace : AllStatic {
 
 class ReservedSpaceRunnable : public TestRunnable {
 public:
-  void runUnitTest() {
+  void runUnitTest() const {
     TestReservedSpace::test_reserved_space();
   }
 };
 
 TEST_VM(VirtualSpace, reserve_space_concurrent) {
-  ConcurrentTestRunner testRunner(new ReservedSpaceRunnable(), 30, 15000);
+  TestRunnable* runnable = new ReservedSpaceRunnable();
+  ConcurrentTestRunner testRunner(runnable, 30, 15000);
   testRunner.run();
+  delete runnable;
 }
 
 class VirtualSpaceRunnable : public TestRunnable {
 public:
-  void runUnitTest() {
+  void runUnitTest() const {
     TestVirtualSpace::test_virtual_space();
   }
 };
 
 TEST_VM(VirtualSpace, virtual_space_concurrent) {
-  ConcurrentTestRunner testRunner(new VirtualSpaceRunnable(), 30, 15000);
+  TestRunnable* runnable = new VirtualSpaceRunnable();
+  ConcurrentTestRunner testRunner(runnable, 30, 15000);
   testRunner.run();
+  delete runnable;
```

Comment:
wouldn't it be easier to allocate TestRunnable on stack and pass a pointer?
```suggestion
  VirtualSpaceRunnable runnable();
  ConcurrentTestRunner testRunner(&runnable, 30, 15000);
  testRunner.run();
```

Reply:
Thanks Igor for these suggestions. I have updated the code accordingly.